eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClientListWithContinuationTokenHeaderResponseCollectionResult.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace BasicTypeSpec
+{
+    internal partial class BasicTypeSpecClientListWithContinuationTokenHeaderResponseCollectionResult : Pageable<BinaryData>
+    {
+        private readonly BasicTypeSpecClient _client;
+        private readonly string _token;
+        private readonly RequestContext _context;
+
+        /// <summary> Initializes a new instance of BasicTypeSpecClientListWithContinuationTokenHeaderResponseCollectionResult, which is used to iterate over the pages of a collection. </summary>
+        /// <param name="client"> The BasicTypeSpecClient client used to send requests. </param>
+        /// <param name="token"></param>
+        /// <param name="context"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        public BasicTypeSpecClientListWithContinuationTokenHeaderResponseCollectionResult(BasicTypeSpecClient client, string token, RequestContext context)
+        {
+            _client = client;
+            _token = token;
+            _context = context;
+        }
+
+        /// <summary> Gets the pages of BasicTypeSpecClientListWithContinuationTokenHeaderResponseCollectionResult as an enumerable collection. </summary>
+        /// <param name="continuationToken"> The continuation token. </param>
+        /// <param name="pageSizeHint"> The page size hint. </param>
+        /// <returns> The pages of BasicTypeSpecClientListWithContinuationTokenHeaderResponseCollectionResult as an enumerable collection. </returns>
+        public override IEnumerable<Page<BinaryData>> AsPages(string continuationToken, int? pageSizeHint)
+        {
+            string nextLink = continuationToken;
+            do
+            {
+                Response response = GetNextResponse(pageSizeHint, nextLink);
+                if (response is null)
+                {
+                    yield break;
+                }
+                using JsonDocument jsonDoc = JsonDocument.Parse(response.Content.ToString());
+                JsonElement root = jsonDoc.RootElement;
+                List<BinaryData> items = new List<BinaryData>();
+                if (root.TryGetProperty("Things", out JsonElement itemsArray))
```

Comment:
Should we use the response model type rather than directly using System.Text.Json?

Reply:
OK, I had some misunderstanding of Protocol methods. Yes, we could just use the response model type directly.